generator/src/main/java/com/scottlogic/deg/generator/utils/JavaUtilRandomNumberGenerator.java

```diff
@@ -42,37 +41,23 @@ public int nextInt(int bound) {
     }
 
     @Override
-    public int nextInt(int lowerInclusive, int upperExclusive) {
-        // implementation copied from Random::internalNextInt
-        if (lowerInclusive < upperExclusive) {
-            int n = upperExclusive - lowerInclusive;
-            if (n > 0) {
-                return nextInt(n) + lowerInclusive;
-            }
-            else {  // range not representable as int
-                int r;
-                do {
-                    r = nextInt();
-                } while (r < lowerInclusive || r >= upperExclusive);
-                return r;
-            }
-        }
-        else {
-            return nextInt();
-        }
+    public long nextLong(long lowerInclusive, long upperExclusive) {
+        long multiplier = (long) (random.nextDouble() * (double) (upperExclusive - lowerInclusive));
+        return multiplier + lowerInclusive;
     }
 
     @Override
     public double nextDouble(double lowerInclusive, double upperExclusive) {
-        return random.nextDouble()
-            * (upperExclusive - lowerInclusive)
+        return (random.nextDouble()
+            * (upperExclusive - lowerInclusive))
             + lowerInclusive;
     }
 
     @Override
     public BigDecimal nextBigDecimal(BigDecimal lowerInclusive, BigDecimal upperExclusive) {
-        return new BigDecimal(random.nextDouble())
+        return BigDecimal.valueOf(random.nextDouble())
```

Comment:
Is this just best practice or is it actually changing functionality?

Reply:
It _could_ change functionality but given it is a random number generator, and that a change in functionality would be more likely to be bounded correctly, this shouldn't be an issue.

Reply:
Not sure I fully understand.

Does `BigDecimal.valueOf(0.1)` return `1` or `0.1000000000000000055511151231257827021181583404541015625?`

Anyway this is more for my interest, I'm happy with the changes.

Reply:
> Does `BigDecimal.valueOf(0.1)` return `1`
Assuming you meant `0.1` here. 🔢 

It returns `0.1`, where `new BigDecimal(0.1)` would not.

`    @Test
    public void testDouble() {
        double value = 0.1;

        assertEquals("0.1", BigDecimal.valueOf(value).toString());
    }`

This test succeeds, if you want to verify.

Reply:
I believe you!